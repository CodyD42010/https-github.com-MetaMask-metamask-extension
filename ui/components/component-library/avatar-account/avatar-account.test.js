/* eslint-disable jest/require-top-level-describe */
import { render } from '@testing-library/react';
import React from 'react';
import { AvatarAccount, AVATAR_ACCOUNT_SIZES } from '.';
import 'jest-canvas-mock';

describe('AvatarAccount', () => {
  const args = {
    address: '0x5CfE73b6021E818B776b421B1c4Db2474086a7e1',
    type: 'Jazzicon',
  };
  it('should render correctly', () => {
    const { getByTestId, container } = render(
      <AvatarAccount data-testid="avatar-account" {...args} />,
    );
    expect(getByTestId('avatar-account')).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  it('should render Jazzicon correctly', () => {
    const container = (
      <AvatarAccount data-testid="avatar-account" {...args} type="Jazzicon" />
    );
    expect(container.props.type).toStrictEqual('Jazzicon');
  });

  it('should render Blockie correctly', () => {
    const container = (
      <AvatarAccount data-testid="avatar-account" {...args} type="Blockie" />
    );
    expect(container.props.type).toStrictEqual('Blockie');
  });

  it('should render with custom classname', () => {
    const { getByTestId } = render(
      <AvatarAccount
        className="mm-avatar-account--test"
        data-testid="classname"
        {...args}
      />,
    );
    expect(getByTestId('classname')).toHaveClass('mm-avatar-account--test');
  });

  it('should render with address', () => {
    const container = (
      <AvatarAccount
        className="mm-avatar-account--test"
        data-testid="classname"
        {...args}
        address="0x0"
      />
    );
    expect(container.props.address).toStrictEqual('0x0');
  });

  it('should render with different size classes', () => {
    const { getByTestId } = render(
      <>
        <AvatarAccount
          size={AVATAR_ACCOUNT_SIZES.XS}
          data-testid={AVATAR_ACCOUNT_SIZES.XS}
          {...args}
        />
        <AvatarAccount
          size={AVATAR_ACCOUNT_SIZES.SM}
          data-testid={AVATAR_ACCOUNT_SIZES.SM}
          {...args}
        />
        <AvatarAccount
          size={AVATAR_ACCOUNT_SIZES.MD}
          data-testid={AVATAR_ACCOUNT_SIZES.MD}
          {...args}
        />
        <AvatarAccount
          size={AVATAR_ACCOUNT_SIZES.LG}
          data-testid={AVATAR_ACCOUNT_SIZES.LG}
          {...args}
        />
        <AvatarAccount
          size={AVATAR_ACCOUNT_SIZES.XL}
          data-testid={AVATAR_ACCOUNT_SIZES.XL}
          {...args}
        />
      </>,
    );
    expect(getByTestId(AVATAR_ACCOUNT_SIZES.XS)).toHaveClass(
      `mm-avatar-base--size-${AVATAR_ACCOUNT_SIZES.XS}`,
    );
    expect(getByTestId(AVATAR_ACCOUNT_SIZES.SM)).toHaveClass(
      `mm-avatar-base--size-${AVATAR_ACCOUNT_SIZES.SM}`,
    );
    expect(getByTestId(AVATAR_ACCOUNT_SIZES.MD)).toHaveClass(
      `mm-avatar-base--size-${AVATAR_ACCOUNT_SIZES.MD}`,
    );
    expect(getByTestId(AVATAR_ACCOUNT_SIZES.LG)).toHaveClass(
      `mm-avatar-base--size-${AVATAR_ACCOUNT_SIZES.LG}`,
    );
    expect(getByTestId(AVATAR_ACCOUNT_SIZES.XL)).toHaveClass(
      `mm-avatar-base--size-${AVATAR_ACCOUNT_SIZES.XL}`,
    );
  });
});
