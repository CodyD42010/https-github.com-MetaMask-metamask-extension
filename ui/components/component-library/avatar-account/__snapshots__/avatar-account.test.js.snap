// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarAccount should render correctly 1`] = `
<div>
  <div
    class="box mm-text mm-avatar-base mm-avatar-base--size-md mm-avatar-account mm-text--body-sm mm-text--text-transform-uppercase box--display-flex box--flex-direction-row box--justify-content-center box--align-items-center box--color-text-default box--background-color-background-alternative box--rounded-full box--border-color-border-default box--border-style-solid box--border-width-1"
    data-testid="avatar-account"
  >
    <div
      class="mm-avatar-account__jazzicon"
    >
      <div
        style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 32px; height: 32px; display: inline-block; background: rgb(249, 108, 1);"
      >
        <svg
          height="32"
          width="32"
          x="0"
          y="0"
        >
          <rect
            fill="#F2CA02"
            height="32"
            transform="translate(1.431306801437213 -0.09488555961102602) rotate(529.4 16 16)"
            width="32"
            x="0"
            y="0"
          />
          <rect
            fill="#FAA200"
            height="32"
            transform="translate(10.974858326048588 -13.039888710538095) rotate(336.9 16 16)"
            width="32"
            x="0"
            y="0"
          />
          <rect
            fill="#C8142C"
            height="32"
            transform="translate(-22.963750275720685 -20.223166935456838) rotate(276.6 16 16)"
            width="32"
            x="0"
            y="0"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
